README.md

```diff
@@ -14,82 +20,154 @@
+  <p>Second-level headings are rare in most Ecmarkdown documents, as
+  <code>&lt;emu-clause&gt;</code> plus first-level headings usually serve better. But
+  in some cases you want a heading that does not show up in the table
+  of contents, so you can use <code>##</code> or <code>###</code> syntax.</p>
 
-Lists are written as a series of lines, each starting with either a number, e.g. `1.`, or a star, e.g. `*`. The first list item's number determines the starting number in the output (via `<ol start="x">`); subsequent list items' numbers are ignored.
+  <emu-grammar>
+    Foo :
+      `bar` Baz
+  </emu-grammar>
+
+  <p>The contents of <code>&lt;emu-grammar&gt;</code>, <code>&lt;emu-production&gt;</code>, <code>&lt;pre&gt;</code>,
+  <code>&lt;code&gt;</code>, <code>&lt;script&gt;</code>, and <code>&lt;style&gt;</code> tags are left alone by Ecmarkdown.</p>
+
+  <emu-clause id="sub-clause">
+    <h1>Foo(<var>p1</var>, <var>p2</var>)</h1>
+    <p>Notice how headers can also use Ecmarkdown inline elements.</p>
+  </emu-clause>
+
+  <aside>This content will *not* be processed by Ecmarkdown.
+    <p>But this content will be, since it&#39;s on a new line.</p>
+  </aside>
+</emu-clause>
+```
+
+## Syntax
+
+### Document-Level Structure
+
+A document generally consists of paragraphs, lists, headers, and HTML blocks. Paragraphs are the default, and are created for lines without special markup at the start of them. Except inside opaque HTML blocks, all use of `<`, `>`, `"`, and `'` are escaped to their corresponding HTML entities.
+
+#### Headers
+
+A header line is one beginning with between one and six `#` signs. It will be transformed into a corresponding tag from the `<h1>` through `<h6>` range. Inside a header you can use inline Ecmarkdown constructs.
+
+#### Lists
+
+Lists are written as a series of lines, each starting with either a number, e.g. `1.`, or a star, e.g. `*`. Inside a list item line you can use inline Ecmarkdown constructs. The first list item's number determines the starting number in the output (via `<ol start="x">`); subsequent list items' numbers are ignored.
 
 Lists can be nested. To do so, use any number of spaces to indent; as long as the number of spaces is consistent, list items will stay together in a nested list.
 
-## Inline Formatting
+#### HTML Blocks
+
+Any line which starts with a block-level HTML tag ([as defined by CommonMark](http://spec.commonmark.org/0.22/#html-blocks), with the addition of `<emu-note>`, `<emu-clause>`, `<emu-intro>`, and `<emu-annex>`) is a HTML block line. Ecmarkdown cannot be used on the line starting a HTML block, but subsequent lines before the closing tag do allow it.
+
+#### Opaque HTML Blocks
 
-Within a fragment, the following can be used:
+The tags `<emu-grammar>`, `<emu-production>`, `<pre>`, `<code>`, `<script>`, and `<style>` are considered opaque. Their entire contents, until their closing tag is seen, are left alone, with no Ecmarkdown processing or HTML escaping.
```

Comment:
Hmm what happens if they don't begin a line? CommonMark testing against `<pre>` says it works anyway.